privacyidea/lib/utils/__init__.py

```diff
@@ -1440,4 +1440,34 @@ def to_list(input):
         return input
     if isinstance(input, set):
         return list(input)
-    return [input]
\ No newline at end of file
+    return [input]
+
+
+def parse_string_to_dict(s, split_char=":"):
+    """
+    This function can parse a string that is formatted like:
+
+       :key1: valueA valueB valueC :key2: valueD valueE
+
+    and return a dict:
+
+       {"key1": ["valueA", "valueB", "valueC"],
+        "key2": ["valueD", "valueE"]
+
+    Note: a whitespace is in the string is separating the values.
+    Thus values can not contain a whitespace.
+
+    :param s: The string that should be parsed
+    :param split_char: The character used for splitting the string
+    :return: the dict
+    """
+    d = {}
+    # create a list like ["key1", "valueA valueB valueC", "key2", "valueD valueE"]
+    li = [x.strip() for x in s.strip().split(split_char) if x]
+    li.reverse()
```

Comment:
You could write this part in a more *pythonic* way:
```
d = {a:b for a,b in zip(li[::2], [[x for x in y.split()] for y in li[1::2]])}
```
You don't need to ``strip()`` the values when splitting with a space since all the spaces will be removed by the ``split()``.

Reply:
If I or anyone else would simply understand that!

Reply:
Ok, now I understand. A bit.
Done in 9f10430a